dask/array/slicing.py

```diff
@@ -395,7 +454,16 @@ def take(outname, inname, blockdims, index, axis=0):
                                            (getitem, ('x', 2), ([7],))],
                                           0),
                          (2, 0, 4, 1))}
+
+    When list is sorted we retain original block structure
+
+    >>> take('y', 'x', [(20, 20, 20, 20)], [1, 3, 5, 47], axis=0)  # doctest: +SKIP
```

Comment:
The term `getitem` would look less pretty in the result.  Has this function signature changed?  Note that this function `take` should probably be renamed to as to avoid collision with your new (and more accurately named) take function.